sympy/physics/mechanics/joint.py

```diff
@@ -1378,3 +1380,431 @@ def _set_linear_velocity(self):
+    .. list-table::
+        :header-rows: 1
+
+        * - Provided ``planar_vectors``
+          - Computed ``rotation_axis``
+        * - ``None``
+          - ``z``
+        * - 1 planar vector parallel to ``x``
+          - ``planar_vectors[0] × y``
+        * - 1 planar vector nonparallel to ``x``
+          - ``x × planar_vectors[0]``
+        * - 2 arbritrary planar vectors
+          - ``planar_vectors[0] × planar_vectors[1]``
+
+    Now the rotation axis is known the missig planar vectors are determined. In
+    the case that there are no planar vectors provided, then the first planar
+    vector is determined as follows (if not provided):
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Determined ``rotation_axis``
+          - Computed ``planar_vectors[0]``
+        * - parallel to ``y``
+          - ``rotation_axis × z``
+        * - nonparallel to ``y``
+          - ``y × rotation_axis``
+
+    The second planar vector is determined as follows (if not provided):
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Computed ``planar_vectors[1]``
+        * - ``rotation_axis × planar_vectors[0]``
+
+    Overall the aim is to get ``z`` as rotation axis and ``x`` and ``y`` as
+    planar vectors, which is also the default.
+
+    """
+
+    def __init__(self, name, parent, child, rotation_coordinate=None,
+                 planar_coordinates=None, rotation_speed=None,
+                 planar_speeds=None, parent_point=None, child_point=None,
+                 parent_interframe=None, child_interframe=None,
+                 planar_vectors=None, rotation_axis=None):
+        self._rotation_axis = rotation_axis
+        self._planar_vectors = planar_vectors
+        coordinates = (rotation_coordinate, planar_coordinates)
+        speeds = (rotation_speed, planar_speeds)
+        super().__init__(name, parent, child, coordinates, speeds,
+                         parent_point, child_point,
+                         parent_interframe=parent_interframe,
+                         child_interframe=child_interframe)
+
+    def __str__(self):
+        return (f'PlanarJoint: {self.name}  parent: {self.parent}  '
+                f'child: {self.child}')
+
+    @property
+    def rotation_coordinate(self):
+        """Generalized coordinate corresponding to the rotation angle."""
+        return self.coordinates[0]
+
+    @property
+    def planar_coordinates(self):
+        """Two generalized coordinates used for the planar translation."""
+        return self.coordinates[1:, 0]
+
+    @property
+    def rotation_speed(self):
+        """Generalized speed corresponding to the angular velocity."""
+        return self.speeds[0]
+
+    @property
+    def planar_speeds(self):
+        """Two generalized speeds used for the planar translation velocity."""
+        return self.speeds[1:, 0]
+
+    @property
+    def rotation_axis(self):
+        """The axis about which the rotation occurs."""
+        return self._rotation_axis
```

Comment:
Do we ever add any sanitisation in the other joint classes? Should we be checking whether the user provides a valid vector here?

Reply:
I think this can be ignored, looks like the initialiser is just temporarily storing the value in `self._rotation_axis` with `self._set_vectors` handling the sanitisation.

Reply:
yep, the entire checking is done in `_set_vectors`. The difficulty is that for the checking some of the basic attributes should be set first. So, the first possibility to start setting those properties is in the `_orient_frames` method.